x-pack/plugins/dataset_quality/server/services/index_stats.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core/server';
+import { rangeQuery } from '@kbn/observability-plugin/server';
+import { DataStreamType } from '../../common/types';
+
+class IndexStatsService {
+  public async getIndexStats(
+    esClient: ElasticsearchClient,
+    type: DataStreamType
+  ): Promise<{
+    doc_count: number;
+    size_in_bytes: number;
+  }> {
+    try {
+      const index = `${type}-*`;
+
+      const indexStats = await esClient.indices.stats({ index });
+      return {
+        doc_count: indexStats._all.primaries?.docs ? indexStats._all.primaries?.docs?.count : 0,
+        size_in_bytes: indexStats._all.primaries?.store
```

Comment:
Do we only want the size of primary shards? If so, it's worth adding this info the the tooltip.

Reply:
Since we are using the dataStream API we are getting the value from [all shards](https://www.elastic.co/guide/en/elasticsearch/reference/current/data-stream-stats-api.html).
```
total_store_size_bytes
(integer) Total size, in bytes, of all shards for the selected data streams.
```

Reply:
To match what is in the table yes, we also include replicas in APM storage explorer (and I think in profiling)